__tests__/ship.test.js

```diff
@@ -0,0 +1,293 @@
+'use strict';
```

Comment:
A note on the format of test cases:
One of the most useful ways to arrange the code is in the format **Arrange - Act - Assert**
- **Arrange** the data for the test,
- **Act** on the method you want to test,
- **Assert** the returns are what you expect them to be.
Following this layout will improve the readability of the tests and help keep the logic more focused